### PR TITLE
GitHub Actions

### PR DESCRIPTION
**Pull request by Bodigrim**
Adds GitHub Actions, because Travis CI will soon become unavailable: Haskell org has spent 1/3 of its total OS credit in a day (https://travis-ci.com/organizations/haskell/plan). 
Inspired by @ethercrow work in https://github.com/haskell/bytestring.
Builds [Ubuntu,MacOS] x GHC [7.10..8.10] + FreeBSD x GHC 8.10. Notice that https://github.com/actions/setup-haskell does not support GHC < 7.10.
Sample run: https://github.com/Bodigrim/unix/actions/runs/375453792.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  pull_request: {}
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest, macOS-latest]
+        ghc: ['8.10', '8.8', '8.6', '8.4', '8.2', '8.0', '7.10']
```
Comment (emilypi) [inline .github/workflows/ci.yml:16]: it'd be nice to add `latest` to this as well, so we can build against head.

**Review by emilypi — APPROVED** [review]

**Comment by Bodigrim** [comment]
Added jobs for Ubuntu x GHC [7.4..7.8], which allows to decomission Travis build entirely. 
Sample run: https://github.com/Bodigrim/unix/actions/runs/386012546
If there will be no comments or suggestions, I'll merge this next week.

**Comment by emilypi** [comment]
Thanks @Bodigrim - let's poke @hvr and see if he has objections.

**Comment by Bodigrim** [comment]
Updated after I learned more GitHub Actions tricks.
@hvr ping 
CC @chessai

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      run: cabal haddock -w /opt/ghc/bin/ghc-${{ matrix.ghc }}
+
+  build:
+    needs: build-old
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ubuntu-latest, macOS-latest]
+        ghc: ['8.10', '8.8', '8.6', '8.4', '8.2', '8.0', '7.10']
```
Comment (chessai) [inline .github/workflows/ci.yml:52]: should 7.10 be in build-old?

Comment (Bodigrim) [reply .github/workflows/ci.yml:52]: `build-old` is for configurations unsupported by `actions/setup-haskell` (and underlying `ghcup`), which is GHC < 7.10.

**Review by chessai — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (chessai) [reply .github/workflows/ci.yml:52]: approved

**Pull request merged**